filebeat/harvester/log.go

```diff
@@ -99,6 +99,13 @@ func (h *Harvester) Harvest() {
 			event.Text = &text
 			event.JSONFields = message.Fields
 			event.EventMetadata = h.config.EventMetadata
+			if message.Fields != nil {
```

Comment:
`if len(message.Fields) > 0` would omit an empty MapStr from the event that is being sent. `len()` also returns 0 if the map itself is nil.